editor/src/components/canvas/controls/select-mode/border-radius-control.tsx

```diff
@@ -0,0 +1,241 @@
+  scale: number
+  color: string
+  elementSize: Size
+  showIndicatorFromParent: boolean
+  showDot: boolean
+}
+
+const CircularHandle = React.memo((props: CircularHandleProp) => {
+  const {
+    borderRadius,
+    isDragging,
+    backgroundShown,
+    scale,
+    color,
+    canvasOffsetRef,
+    dispatch,
+    corner,
+    elementSize,
+    showIndicatorFromParent,
+    showDot,
+  } = props
+
+  const [hovered, setHovered] = React.useState<boolean>(false)
+
+  const handleHoverStart = React.useCallback(() => setHovered(true), [])
+  const handleHoverEnd = React.useCallback(() => setHovered(false), [])
+
+  const handleMouseDown = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) =>
+      startInteraction(e, dispatch, canvasOffsetRef.current, scale, corner),
+    [canvasOffsetRef, corner, dispatch, scale],
+  )
+
+  const shouldShowIndicator = (!isDragging && hovered) || showIndicatorFromParent
+  const shouldShowHandle = isDragging || backgroundShown
+
+  const size = BorderRadiusHandleSize(scale)
+  const position = handlePosition(borderRadius.renderedValuePx, elementSize, corner, scale)
+
+  return (
+    <div
+      data-testid={CircularHandleTestId(corner)}
+      style={{
+        position: 'absolute',
+        left: position.x,
+        top: position.y,
+      }}
+      onMouseEnter={handleHoverStart}
+      onMouseLeave={handleHoverEnd}
+      onMouseDown={handleMouseDown}
+    >
+      <>
+        {when(
+          shouldShowIndicator,
+          <div
+            style={{
+              position: 'absolute',
+              paddingLeft: BorderRadiusHandleHitArea(scale),
+              paddingTop: BorderRadiusHandleHitArea(scale),
+              pointerEvents: 'none',
+            }}
+          >
+            <CSSNumberLabel
+              prefix='Radius'
+              value={borderRadius.value}
+              scale={scale}
+              color={color}
+            />
+          </div>,
+        )}
+        <div
+          style={{
+            visibility: shouldShowHandle ? 'visible' : 'hidden',
+            width: size,
+            height: size,
+            backgroundColor: 'white',
+            border: `${BorderRadiusHandleBorderWidth(scale)}px solid blue`,
+            borderRadius: '50%',
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'center',
+          }}
+        >
+          <div
+            style={{
+              visibility: shouldShowHandle && showDot ? 'visible' : 'hidden',
+              width: BorderRadiusHandleDotSize(scale),
+              height: BorderRadiusHandleDotSize(scale),
+              backgroundColor: 'blue',
+              borderRadius: '50%',
+            }}
+          />
+        </div>
+      </>
+    </div>
+  )
+})
+
+function startInteraction(
+  event: React.MouseEvent<HTMLDivElement>,
+  dispatch: EditorDispatch,
+  canvasOffset: CanvasVector,
+  scale: number,
+  corner: BorderRadiusCorner,
+) {
+  event.stopPropagation()
```

Comment:
This should probably be inside the `if` statement below

Reply:
Double check that though incase it causes selection issues (I don't _think_ it will because of my recent mouseup fix, but there might be something else we've missed)

Reply:
done!